### PR TITLE
JAMES-2555 Make ReIndexing APIs more REST friendly

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/MessageIdReindexingRoutes.java [hunk 1]
```diff
@@ -50,19 +48,20 @@
 import spark.Service;
 
 @Api(tags = "MessageIdReIndexing")
-@Path("/mailboxIndex")
+@Path("/messages")
```
Comment (aduprat) [inline server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/MessageIdReindexingRoutes.java]: I think you have to modify the `@Path` on line 82, else you are going to introduce a `/messages/messages/{messageId}` endpoint.

**Comment by aduprat** [comment]
I afraid about such modifications, clients will be broken.

Can we think about introducing Deprecated annotation on old endpoints & keep them?

**Comment by chibenwa** [comment]
`SpamAssassinContract.movingAMailToTrashShouldNotImpactSpamassassinLearning:370 » ConditionTimeout`

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by blackheaven** [comment]
I agree, how do you plan to do it? (I mean, will you review all the commits or do you have already a list)

**Comment by chibenwa** [comment]
> I agree, how do you plan to do it? (I mean, will you review all the commits or do you have already a list)

Review all the PRs

**Comment by chibenwa** [comment]
test this pease

**Comment by trantienduchn** [comment]
```
[6ac06c97f50f069dcaa06f69ba5fa1292f605213] [ERROR] Errors: 
[6ac06c97f50f069dcaa06f69ba5fa1292f605213] [ERROR]   FilterTest.messageShouldBeAppendedInInboxWhenFromDoesNotMatchRule:1068 » JsonPath
[6ac06c97f50f069dcaa06f69ba5fa1292f605213] [ERROR]   FilterTest.setup:79 » Provision Unable to provision, see the following errors:...
[6ac06c97f50f069dcaa06f69ba5fa1292f605213] [ERROR]   RabbitMQGetMessagesMethodTest » ConditionTimeout Condition with lambda express...
[6ac06c97f50f069dcaa06f69ba5fa1292f605213] [ERROR]   RabbitMQGetMessagesMethodTest » ConditionTimeout Condition with lambda express...
[6ac06c97f50f069dcaa06f69ba5fa1292f605213] [ERROR]   RabbitMQGetMessagesMethodTest » ConditionTimeout Condition with lambda express...
[6ac06c97f50f069dcaa06f69ba5fa1292f605213] [ERROR]   RabbitMQMailboxSharingTest » ConditionTimeout Lambda expression in org.apache....
```
test this please

**Comment by chibenwa** [comment]
:green_apple: see https://github.com/linagora/james-project/pull/1993

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**